File: sdks/node/src/domain/core/cyan.ts
enum GlobType {
  Template,
  Copy,
}

interface CyanGlob {
  root?: string | null;
  glob: string;
  exclude: string[];
  type: GlobType;
}

interface CyanPlugin {
  name: string;
  config: unknown;
}

interface CyanProcessor {
  name: string;
  files: CyanGlob[];
  config: unknown;
}

interface Cyan {
  processors: CyanProcessor[];
  plugins: CyanPlugin[];
}

// export all
export { GlobType };
export type { Cyan, CyanProcessor, CyanPlugin, CyanGlob };

File: sdks/python/cyanprintsdk/domain/core/cyan.py
from dataclasses import dataclass
from enum import Enum
from typing import Optional, List, Any


@dataclass
class GlobType(Enum):
    Template = 1
    Copy = 2


@dataclass
class CyanGlob:
    glob: str
    type: GlobType
    root: Optional[str] = None
    exclude: Optional[List[str]] = None


@dataclass
class CyanPlugin:
    name: str
    config: Any


@dataclass
class CyanProcessor:
    name: str
    files: List[CyanGlob]
    config: Any


@dataclass
class Cyan:
    processors: List[CyanProcessor]
    plugins: List[CyanPlugin]
